docker/Dockerfile.aarch64-linux-android

```diff
@@ -63,4 +69,7 @@ ENV CARGO_TARGET_AARCH64_LINUX_ANDROID_LINKER="$CROSS_TOOLCHAIN_PREFIX"gcc \
     TMPDIR=/tmp/ \
     ANDROID_DATA=/ \
     ANDROID_DNS_MODE=local \
-    ANDROID_ROOT=/system
+    ANDROID_ROOT=/system \
+    CROSS_CMAKE_SYSTEM_NAME=Android \
+    CROSS_CMAKE_SYSTEM_PROCESSOR=aarch64 \
+    CROSS_CMAKE_CRT=android
```

Comment:
`CROSS_CMAKE_CRT` is technically not needed for this: it's only needed for `newlib` and we could probably do `CROSS_CMAKE_NEWLIB` but I figured it might be nice just to have this. It might be better to check for `*-none-*` though since we only use it to check if we should build an executable or staticlib when testing if the compiler works.